### PR TITLE
Adds more splat and tweaks PuddleComponent.cs

### PR DESCRIPTION
**Pull request by SweptWasTaken**
## About the PR 

PuddleComponent.cs had hard-coded sprite states that was supplied by the RSI file name? Was sus so I fixed it and updated all the puddle components in yaml.

I also added flour and tomato splatting.

**Screenshots**

image

**Changelog**

:cl: Swept
- add: You can splat bags of flour and tomatoes.

### CODE REVIEW

File: Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs [hunk 1]
```diff
@@ -122,10 +125,7 @@ public override void Initialize()
 
             if (_spriteComponent.BaseRSIPath != null)
             {
-                var baseName = new ResourcePath(_spriteComponent.BaseRSIPath).FilenameWithoutExtension;
-
-                _spriteComponent.LayerSetState(0, $"{baseName}-{randomVariant}"); // TODO: Remove hardcode
-
+                _spriteComponent.LayerSetState(0, $"{_spriteState}-{randomVariant}"); // TODO: Remove hardcode
```
Comment (mirrorcult) [inline Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs:128]: do you need the comment still?

Comment (PaulRitter) [reply Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs:128]: no

File: Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml [hunk 2]
```diff
@@ -214,6 +214,39 @@
...
+        damage: 1
+      behaviors:
+      - !type:PlaySoundCollectionBehavior
+        soundCollection: desecration
+      - !type:SpawnEntitiesBehavior
+        spawn:
+          PuddleTomato:
+            min: 1
+            max: 1
+          # Wow double-yolk you're so lucky!
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml:235]:
```suggestion
```

File: Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs (hunk 1 above)
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs:128]:
```suggestion
                _spriteComponent.LayerSetState(0, $"{_spriteState}-{randomVariant}");
```

**Review by PaulRitter — CHANGES_REQUESTED** [review]
looks good

**Comment by SweptWasTaken** [comment]
Just direct-applied the reviews Paul left, will self-merge if I come back from breakfast and the tests are passing.

**Pull request merged**